k8s/names.go

```diff
@@ -26,13 +25,22 @@ func ToSubdomain(in string) (string, error) {
 			out = append(out, c)
 		}
 	}
-	return string(out), nil
+
+	result := string(out)
+
+	if errs := validation.IsDNS1035Label(result); len(errs) > 0 {
```

Comment:
What kinds of characters in `in` will cause this to fail?

Reply:
Added comment with a description